### ISSUE TITLE
Update react-router 0.13.3 -> 1.0.0

### ISSUE THREAD

**Issue by GiedriusGrabauskas** [DefinitelyTyped/DefinitelyTyped#5939]
Upgrade guide:
https://github.com/rackt/react-router/blob/master/UPGRADE_GUIDE.md
Glossary:
https://github.com/rackt/react-router/blob/master/docs/Glossary.md

**Comment by stepancar** [comment]
+1, and Please add es6 import support
see router  documentation
```javascript
import { Router, Route, Link } from 'react-router'
```

**Comment by sergey-buturlakin** [comment]
Hi,
New react-router uses new standalone history module. I've created complete definitions for it and also simple definitions (for basic usage, es6 compatible) for new react-router.
It would be good to use these definitions in the main repo.

**Comment by sergey-buturlakin** [comment]
https://github.com/sergey-buturlakin/DefinitelyTyped/tree/master/react-router-history

**Comment by ccapndave** [comment]
+1
Can you make a pull request?

**Comment by sergey-buturlakin** [comment]
There is small issue with <code>history</code> module naming. According to the author's docs https://github.com/rackt/history#readme the module isn't bounded to react and could be used in any JS project. In <code>npm</code> it is named as <code>history</code>. So, logically, it should be named as history in the main repo. But there is already another <code>history</code> defs in the main repo https://github.com/borisyankov/DefinitelyTyped/tree/master/history.
So, what name to use for the new definitions?

**Comment by ccapndave** [comment]
As a workaround you could specify the functions individually (this is what I have been doing):

```
declare module "history/lib/createHashHistory" {
    export default function createHashHistory(): any;
}

declare module "history/lib/createBrowserHistory" {
    export default function createBrowserHistory(): any;
}

declare module "history/lib/createMemoryHistory" {
    export default function createMemoryHistory(): any;
}
```

**Comment by stepancar** [comment]
@sergey-buturlakin , thank you for this react-router definitions!!!
 it's what i neads.
make pull request please. i think your definitions is good

**Comment by arusakov** [comment]
@sergey-buturlakin thanks, waiting for PR!

**Comment by sergey-buturlakin** [comment]
I've made the update, added more definitions for the new react-router:

https://github.com/sergey-buturlakin/DefinitelyTyped/tree/master/react-router

I propose to place history module used by react-router into react-router dir, since history folder in the repository root already exists (historyjs library definitions).

P.S. I need to complete some definitions before PR, but existing defitions are ready for standard usage

P.P.S. I've tested the definitions only with CommonJS modules, maybe somebody tried in different scenarios (global, amd ...)?
